File: Algorithms_python/59_Spiral_Matrix_II.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        res = [[0] * n for _ in range(n)]
        num = 1
        m1, m2 = 0, n-1
        while m1 < m2:
            for i in range(m1, m2):
                res[m1][i] = num
                num += 1
            for j in range(m1, m2):
                res[j][m2] = num
                num += 1
            for i in range(m2, m1, -1):
                res[m2][i] = num
                num += 1
            for j in range(m2, m1, -1):
                res[j][m1] = num
                num += 1
            m1 += 1
            m2 -= 1
        if m1 == m2:
            res[m1][m2] = num
        return res
'''
没有什么好解法，直接暴力循环，从外层向内层循环，注意好数组标识顺序就行，
别忘了最后一个中间数
这个写法很清楚，就没有做新版改进
'''

File: Algorithms_C#/59_Spiral_Matrix_II.cs
public class Solution {
    public int[][] GenerateMatrix(int n) {
        int[][] ans = new int[n][];
        for (int a = 0; a < n; ++a){
            ans[a] = new int[n];
        }
        int[][] dirs = {new int[]{0, 1}, new int[]{1, 0}, new int[]{0, -1}, new int[]{-1, 0}};
        int i = 0, j = 0, k = 0;
        for (int v = 1; v <= n * n; ++v){
            ans[i][j] = v;
            int x = i + dirs[k][0], y = j + dirs[k][1];
            if (x < 0 || y < 0 || x >= n || y >= n || ans[x][y] != 0){
                k = (k + 1) % 4;
                x = i + dirs[k][0];
                y = j + dirs[k][1];
            }
            i = x;
            j = y;
        }
        return ans;
    }
}